src/internal/kopia/data_collection.go

```diff
@@ -35,6 +41,25 @@ func (kdc kopiaDataCollection) FullPath() path.Path {
 	return kdc.path
 }
 
+func (kdc kopiaDataCollection) Fetch(
+	ctx context.Context,
+	name string,
+) (data.Stream, error) {
+	if kdc.snapshotRoot == nil {
+		return nil, clues.New("no snapshot root")
+	}
+
+	p, err := kdc.FullPath().Append(name, true)
+	if err != nil {
+		return nil, clues.Wrap(err, "creating item path")
+	}
+
+	// TODO(ashmrtn): We could possibly hold a reference to the folder this
```

Comment:
Wondering what the use would be to storing the reference to the folder.

Reply:
it really depends on how we want to play things. Right now we store only the root and require kopia to traverse all directories on the path of an item if we call `Fetch`. If we stored the directory reference for the collection then kopia would only need to access the data in the held reference instead of traversing a tree

The tradeoff is that we'd end up holding references to as many directories as we're restoring instead of holding a single reference to the root and temporarily holding references to directories as we traverse them